docs/root/configuration/listener_filters/http_inspector.rst

```diff
@@ -0,0 +1,37 @@
+.. _config_listener_filters_http_inspector:
+
+HTTP Inspector
+==============
+
+HTTP Inspector listener filter allows detecting whether the application protocol appears to be HTTP, 
+and if it is HTTP, it detects the HTTP protocol (HTTP/1.x or HTTP/2) further. This can be used to select a
+:ref:`FilterChain <envoy_api_msg_listener.FilterChain>` via the :ref:`application_protocols <envoy_api_field_listener.FilterChainMatch.application_protocols>`
+of a :ref:`FilterChainMatch <envoy_api_msg_listener.FilterChainMatch>`.
+
+* :ref:`v2 API reference <envoy_api_field_listener.ListenerFilter.name>`
+* This filter should be configured with the name *envoy.listener.http_inspector*.
+
+Example
+-------
+
+A sample filter configuration could be:
+
+.. code-block:: yaml
+
+  listener_filters:
+    - name: "envoy.listener.http_inspector"
+      typed_config: {}
+
+Statistics
+----------
+
+This filter has a statistics tree rooted at *http_inspector* with the following statistics: 
+
+.. csv-table::
+  :header: Name, Type, Description
+  :widths: 1, 1, 2
+
+  read_error, Counter, Total read errors
+  http1x_found, Counter, Total number of times HTTP/1.x was found
```

Comment:
You should either split this into `http10` and `http11` (which might be useful for tracking legacy clients) or just make it `http1` (since that's how it's referred in Envoy in most of the other places).

Reply:
Split `http1x` into `http10` and `http11`.